### PR TITLE
Migrate network settings to venus-platform

### LINKED ISSUE #618: Ethernet status shows "unplugged" when ethernet connection is manually configured
Reported issue: in ethernet settings, ethernet status shows "unplugged" when ethernet connection is manually configured. 

Status displays correctly in gui-v1.

### PR DESCRIPTION
**Pull request by jpetrell**
The functionality is limited (no way to sign in to hidden networks, no WPA certs, identity, domain, etc.), but should now match the existing functionality. 

Fixes #618.

### CODE REVIEW

**Comment by jpetrell** [comment]
Could not yet drop Connman dependency altogether, filed https://github.com/victronenergy/gui-v2/issues/931 for the remaining work.

**Review by blammit — COMMENTED** [review]
@jpetrell are you able to see wifi networks? With gui-v1 I can go to Wifi Settings and click "Wi-Fi networks" and see the network I'm currently connected to, but with gui-v2 on the same device via wasm, it doesn't show the connected network, or any other network.

File: pages/settings/PageSettingsWifi.qml [hunk 1]
```diff
@@ -10,103 +10,134 @@ import Victron.VenusOS
...
+				model.clear()
+				return
+			}
+
+			var wifis = JSON.parse(value)["wifi"]
 
-	onIsCurrentPageChanged: {
-		if (isCurrentPage) {
-			_reload()
+			var services = Object.values(wifis)
```
[the remark this answers is not in the document]
Comment (jpetrell) [reply pages/settings/PageSettingsWifi.qml:32]: Yeah. Fixed.

File: pages/settings/PageSettingsTcpIp.qml [hunk 2]
```diff
@@ -9,100 +9,95 @@ import Victron.VenusOS
...
-		return service.ipv4[name] || "--"
+	property string network: "Wired"
+	property string tech: "ethernet"
+
+	property alias service: service.service
+	readonly property bool ready: root.service.length > 0
+
+	readonly property bool _wifi: tech === "wifi"
+
+	function action(action) {
```
[the remark this answers is not in the document]
Comment (jpetrell) [reply pages/settings/PageSettingsTcpIp.qml:20]: Fixed.

**Comment by jpetrell** [comment]
> @jpetrell are you able to see wifi networks? With gui-v1 I can go to Wifi Settings and click "Wi-Fi networks" and see the network I'm currently connected to, but with gui-v2 on the same device via wasm, it doesn't show the connected network, or any other network.

Testing this requires updated venus-platform binary. Sorry totally forgot to mention this in the PR!

**Comment by jpetrell** [comment]
Changing the PR to draft as cannot be merged yt. Pending the integration of the respective venus-platform changes.

**Comment by jpetrell** [comment]
Dropped the usability improvements, moved to a separate PR  #953.

**Comment by jpetrell** [comment]
Migrated to the latest changes introduced on the Venus side:
- Wi-Fi settings call setAgent("on"/"off") on page opening/closing
- API provides state string as ConnMan state instead of Venus states (e.g. "idle" vs. "Disconnected")
- Use JSON.stringify() to clean up JSON handling

**Comment by jpetrell** [comment]
Fixed merge conflict.

File: pages/settings/PageSettingsTcpIp.qml [hunk 3]
```diff
@@ -9,100 +9,110 @@ import Victron.VenusOS
...
-		target: Connman
-		function onServiceRemoved(path) {
-			if (path === root.path) {
-				root.path = ""
+		function parseJson() {
+			if (!isValid || typeof value !== "string") {
+				return
+			}
+
+			var services = JSON.parse(value)
```
Comment (blammit) [inline pages/settings/PageSettingsTcpIp.qml:74]: Prefer `const` or `let`? Here and below.

File: pages/settings/PageSettingsTcpIp.qml [hunk 4]
```diff
@@ -180,23 +194,24 @@ Page {
...
-		ListTextItem {
+		ListQuantityItem {
 			text: CommonWords.signal_strength
-			secondaryText: root.service ? service.strength + "%" : ""
+			value: service.strength
+			unit: VenusOS.Units_Percentage
 			allowed: root._wifi
 		}
 
+		// TODO: Report MAC address (BSSID) of Wi-Fi networks
```
Comment (blammit) [inline pages/settings/PageSettingsTcpIp.qml:209]: Could you create a task for this if there isn't already? And put the issue number in here?

File: pages/settings/PageSettingsTcpIp.qml [hunk 5]
```diff
@@ -217,56 +232,59 @@ Page {
...
 				}
 				return -1
 			}
+
+			enabled: userHasReadAccess
+			allowed: defaultAllowed && currentIndex >= 0
+
 			onOptionClicked: function(index) {
-				root._setMethod(optionModel[index].value)
+				setProperty("Method", optionModel[index].value)
```
Comment (blammit) [inline pages/settings/PageSettingsTcpIp.qml:246]: Could the function name be less generic, e.g. `setServiceProperty`? Same with `setAction`.

File: pages/settings/PageSettingsWifi.qml [hunk 6]
```diff
@@ -10,103 +10,134 @@ import Victron.VenusOS
...
+			.filter((object) => object && typeof object === "object")
```
Comment (blammit) [inline pages/settings/PageSettingsWifi.qml:33]: Add indentation to this line and the next?

File: pages/settings/PageSettingsWifi.qml [hunk 7]
```diff
@@ -10,103 +10,134 @@ import Victron.VenusOS
...
-	}
+		onValueChanged: if (root.isCurrentPage) update()
+
+		function update() {
+			if (!isValid || !scanItem.isValid) {
+				model.clear()
+				return
+			}
+
+			var wifis = JSON.parse(value)["wifi"]
```
Comment (blammit) [inline pages/settings/PageSettingsWifi.qml:30]: The json is not trivial to parse. Would it be worthwhile to have a comment showing one of the json entries that Rein posted in the task, to document the data that is being parsed here?

**Review by blammit — APPROVED** [review]
Great work, nice to have this working from MQTT now :)

File: pages/settings/PageSettingsWifi.qml (hunk 6 above)
Comment (jpetrell) [reply pages/settings/PageSettingsWifi.qml:33]: Done.

File: pages/settings/PageSettingsTcpIp.qml (hunk 5 above)
Comment (jpetrell) [reply pages/settings/PageSettingsTcpIp.qml:246]: Fixed.

File: pages/settings/PageSettingsTcpIp.qml (hunk 4 above)
Comment (jpetrell) [reply pages/settings/PageSettingsTcpIp.qml:209]: There exists Venus side issue, mentioning it now in the comments.

File: pages/settings/PageSettingsTcpIp.qml (hunk 3 above)
Comment (jpetrell) [reply pages/settings/PageSettingsTcpIp.qml:74]: Fixed.

File: pages/settings/PageSettingsWifi.qml (hunk 7 above)
Comment (jpetrell) [reply pages/settings/PageSettingsWifi.qml]: Good idea. Added.

**Pull request merged**